src/amber/pipes/client_ip.cr

```diff
@@ -0,0 +1,34 @@
+module Amber
+  module Pipe
+    # Handler to determine real client IP address when Amber is behind a
+    # reverse proxy and save it to context.client_ip.
+    #
+    # The proxy should pass client IP in a header such as "X-Forwarded-For".
+    # In case of multiple values in a header, the last one is used.
+    #
+    # Crystal does not currently make remote IP available, so it is impossible
+    # to accept the header value only when coming from trusted proxy IPs.
+    # For that and other benefits, it is suggested to always run Amber apps
+    # behind proxies such as HAProxy or Nginx.
+    class ClientIp < Base
+      def initialize(header : String = "X-Forwarded-For")
+        @headers = [header]
+      end
+      def initialize( @headers : Array(String))
+      end
+
+      def call(context : HTTP::Server::Context)
+        @headers.each do |header|
+          if addresses = context.request.headers.get?(header)
+            context.client_ip = Socket::IPAddress.new(addresses[-1], 0)
```

Comment:
I think you want `addresses[0]`.  The first is usually the `client_ip`.  Subsequent IP addresses are the proxies.

Reply:
You're right about this one. I've misunderstood the documentation.